src/helpers.js

```diff
@@ -115,3 +115,20 @@ export function makePositionIndexHook(posIndex) {
     return node;
   };
 }
+
+const langToMimeModesMapping = {
+  css: 'text/css',
+  ecmascript: 'text/ecmascript',
+  html: 'text/html',
+  javascript: 'text/javascript',
+  typescript: 'text/typescript',
+  cpp: 'text/x-c++src',
+  xml: 'text/xml',
+  bash: 'text/x-sh',
+  shell: 'text/x-sh'
+};
```

Comment:
It's not true. Bblfsh returns `name` as a human-readable string (JavaScript) and `id` the value of the language you must pass to it for parsing (javascript). This function accepts `id` of course same as right now we pass `id` as `langMode`.

I don't get your second point too. We have all beta languages for sure and many others. As you can see functions returns:
```
langToMimeModesMapping[lang] || `text/x-${lang}`
```

The map has only exceptional cases. For other languages, code-mirror expects `text/x-lang`.

The full list of modes we bundle library with:
```
"application/ecmascript",​
"application/javascript",​
"application/json",​
"application/ld+json",​
"application/typescript",​
"application/x-httpd-php",​
"application/x-httpd-php-open",​
"application/x-javascript",​
"application/x-json",​
"application/x-sh",​
"application/xml",​
"text/css",​
"text/ecmascript",​
"text/html",​
"text/javascript",​
"text/typescript",​
"text/x-c",​
"text/x-c++hdr",​
"text/x-c++src",​
"text/x-cassandra",​
"text/x-ceylon",​
"text/x-chdr",​
"text/x-csharp",​
"text/x-csrc",​
"text/x-cython",​
"text/x-esper",​
"text/x-go",​
"text/x-gpsql",​
"text/x-gql",​
"text/x-gss",​
"text/x-hive",​
"text/x-java",​
"text/x-kotlin",​
"text/x-less",​
"text/x-mariadb",​
"text/x-mssql",​
"text/x-mysql",​
"text/x-nesc",​
"text/x-objectivec",​
"text/x-pgsql",​
"text/x-php",​
"text/x-plsql",​
"text/x-python",​
"text/x-ruby",​
"text/x-scala",​
"text/x-scss",​
"text/x-sh",​
"text/x-sparksql",​
"text/x-sql",​
"text/x-sqlite",​
"text/x-squirrel",​
"text/xml",​
"x-shader/x-fragment",​
"x-shader/x-vertex"
```

there are might be some other exceptions but because we don't support highlighting for them there is no reason to include them in the mapping.